aspnetcore/mvc/advanced/custom-model-binding/3.0sample/PolymorphicModelBinding/Pages/AddDevice.cshtml.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.Extensions.Logging;
+using PolymorphicModelBinding.ModelBinders;
+
+namespace PolymorphicModelBinding.Pages
+{
+    public class AddDeviceModel : PageModel
+    {
+        private readonly ILogger<IndexModel> _logger;
+
+        public AddDeviceModel(ILogger<IndexModel> logger)
+        {
+            _logger = logger;
+        }
+
+        public List<SelectListItem> DeviceKinds { get; } = new List<SelectListItem>
+        {
+            new SelectListItem { Text = "Laptop", },
+            new SelectListItem { Text = "SmartPhone" },
+            new SelectListItem { Text = "Tablet" },
```

Comment:
This is an invalid kind, but it demonstrates how the model-binder behaves when presented with an unknown type.